src/models/node.js

```diff
@@ -919,6 +931,37 @@ const Node = {
       }, [])
   },
 
+  getActiveMarksAtRangeAsArray(range) {
+    range = range.normalize(this)
+    const { startKey, startOffset } = range
+
+    // If the range is collapsed at the start of the node, check the previous.
+    if (range.isCollapsed && startOffset == 0) {
+      const previous = this.getPreviousText(startKey)
+      if (!previous || !previous.length) return []
+      const char = previous.characters.get(previous.length - 1)
+      return char.marks.toArray()
+    }
+
+    // If the range is collapsed, check the character before the start.
+    if (range.isCollapsed) {
+      const text = this.getDescendant(startKey)
+      const char = text.characters.get(range.startOffset - 1)
+      return char.marks.toArray()
+    }
+
+    // Otherwise, get a set of the marks for each character in the range.
+    const chars = this.getCharactersAtRange(range)
+    const first = chars.first()
+    if (!first) return []
```

Comment:
Gotcha, can you take it out for now? I'd like to just keep it terse, and if it becomes an issue we can solve the underlying problem. Thanks!